### PR TITLE
clothing_tint component

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
aaaaaaaaaaa i dont know how I didn't notice how badly tinting was done.

Closes #3385

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -279,7 +279,12 @@
 // for items that can be placed in multiple slots
 // note this isn't called during the initial dressing of a player
 /obj/item/proc/equipped(mob/user, slot)
+	SHOULD_CALL_PARENT(TRUE) // no exceptions
 	SEND_SIGNAL(src, COMSIG_ITEM_EQUIPPED, user, slot)
+	if(flags_equip_slot & slotdefine2slotbit(slot)) // flags_equip_slot is a bitfield
+		SEND_SIGNAL(src, COMSIG_ITEM_EQUIPPED_TO_SLOT, user)
+	else
+		SEND_SIGNAL(src, COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT, user, slot)
```
Comment (Rohesie) [inline code/game/objects/items.dm]: Do a single signal and make it pass if equipped or not as a variable.

Comment (spookydonut) [reply code/game/objects/items.dm]: That would completely defeat the purpose of these signals

Comment (Rohesie) [reply code/game/objects/items.dm]: No, you can still register to the signal and check for the var.

Comment (spookydonut) [reply code/game/objects/items.dm]: That's what you do with the current signal so you're asking me to just make a duplicate signal

Comment (ninjanomnom) [reply code/game/objects/items.dm]: normally I wouldn't be too much a fan of that kind of thing but there's nothing inherently wrong with it and in that case in particular that check is done in a lot of component code. In this context this is a prety nice pair of signals.

File: code/datums/components/clothing_tint.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		return COMPONENT_INCOMPATIBLE
+	if(!isnull(tint))
+		src.tint = tint
+	if(!isnull(tint_state))
+		src.tint_state = tint_state
+
+/datum/component/clothing_tint/Destroy(force, silent)
+	remove_tint()
+	return ..()
+
```
Comment (ninjanomnom) [inline code/datums/components/clothing_tint.dm]: You need to clean up `tinted_mob`

File: code/modules/codex/entries/clothing_codex.dm [hunk 3]
```diff
@@ -79,7 +79,7 @@ GLOBAL_LIST_INIT(string_equip_flags, list("suit slot" = ITEM_SLOT_OCLOTHING,
 	if(!slowdown)
 		armor_strings += "<br>It will not slow the wearer down."
 
-	if(tint)
+	if(GetComponent(/datum/component/clothing_tint))
```
Comment (ninjanomnom) [inline code/modules/codex/entries/clothing_codex.dm]: get_mechanics_info signal

**Pull request merged**